src/tracer.c

```diff
@@ -231,6 +231,27 @@ handle_open(PROCESS_INFO *pi, int fd, int dirfd, void *path, int purpose)
     }
 }
 
+static void
+handle_execve(PROCESS_INFO *pi, int dirfd, char *path)
+{
+    record_process_start(pi->pid, pi->outname);
+
+    char *abspath = absolutepath(pi->pid, dirfd, path);
+    char *hash = get_file_hash(abspath);
+
+    if (!find_finfo(abspath, hash)) {
+	FILE_INFO *f = next_finfo();
+
+	finfo_new(f, path, abspath, hash);
+	record_hash(f->outname, f->hash);
+	f->was_hash_printed = 1;
+    } else {
+	free(abspath);
+	free(hash);
+	free(path);
+    }
```

Comment:
It also needs to record the connection of the file to the process.
I mean, files are connected via `b:reads` or `b:writes` but in this case it should be something like `b:executable`

Reply:
@zvr I actually thought about it but it wasn't mentioned anywhere so i assumed we didn't care.

I was thinking something like `b:execs` but `b:executable` is fine as well.